src/Orchard.Environment.Shell.Abstractions/Builders/Models/ShellBlueprint.cs

```diff
@@ -12,10 +14,25 @@ namespace Orchard.Environment.Shell.Builders.Models
     /// </summary>
     public class ShellBlueprint
     {
+        private static readonly IFeatureInfo CoreFeature
+            = new InternalFeatureInfo("Core", new InternalExtensionInfo("Core"));
+
         public ShellSettings Settings { get; set; }
         public ShellDescriptor Descriptor { get; set; }
 
         public IEnumerable<DependencyBlueprint> Dependencies { get; set; }
+
+        public IFeatureInfo GetFeatureForDependency(Type dependency)
+        {
+            var item = Dependencies.FirstOrDefault(x => x.Type == dependency);
+
+            if (item != null)
+            {
+                return item.Feature;
+            }
+
+            return CoreFeature;
```

Comment:
@jtkech something the might not be clear for you is that even though a service might be hosted in a "core" project, it will be registered by a module in a `Startup.cs`. This file defines where the service is coming from, and if the feature needs to be enabled or not. There might be some outcasts, but we should not care, or list them when we find them as they might show some actual issue in the module architecture.